staging/src/k8s.io/apiextensions-apiserver/pkg/apihelpers/helpers.go

```diff
@@ -54,18 +56,233 @@ const (
 
 // GetAPIApprovalState returns the state of the API approval and reason for that state
 func GetAPIApprovalState(annotations map[string]string) (state APIApprovalState, reason string) {
-	annotation := annotations[v1beta1.KubeAPIApprovedAnnotation]
+	annotation := annotations[apiextensions.KubeAPIApprovedAnnotation]
 
 	// we use the result of this parsing in the switch/case below
 	url, annotationURLParseErr := url.ParseRequestURI(annotation)
 	switch {
 	case len(annotation) == 0:
-		return APIApprovalMissing, fmt.Sprintf("protected groups must have approval annotation %q, see https://github.com/kubernetes/enhancements/pull/1111", v1beta1.KubeAPIApprovedAnnotation)
+		return APIApprovalMissing, fmt.Sprintf("protected groups must have approval annotation %q, see https://github.com/kubernetes/enhancements/pull/1111", apiextensions.KubeAPIApprovedAnnotation)
 	case strings.HasPrefix(annotation, "unapproved"):
 		return APIApprovalBypassed, fmt.Sprintf("not approved: %q", annotation)
 	case annotationURLParseErr == nil && url != nil && len(url.Host) > 0 && len(url.Scheme) > 0:
 		return APIApproved, fmt.Sprintf("approved in %v", annotation)
 	default:
-		return APIApprovalInvalid, fmt.Sprintf("protected groups must have approval annotation %q with either a URL or a reason starting with \"unapproved\", see https://github.com/kubernetes/enhancements/pull/1111", v1beta1.KubeAPIApprovedAnnotation)
+		return APIApprovalInvalid, fmt.Sprintf("protected groups must have approval annotation %q with either a URL or a reason starting with \"unapproved\", see https://github.com/kubernetes/enhancements/pull/1111", apiextensions.KubeAPIApprovedAnnotation)
 	}
 }
+
+var swaggerMetadataDescriptions = metav1.ObjectMeta{}.SwaggerDoc()
+
+// SetCRDCondition sets the status condition. It either overwrites the existing one or creates a new one.
+func SetCRDCondition(crd *apiextensions.CustomResourceDefinition, newCondition apiextensions.CustomResourceDefinitionCondition) {
```

Comment:
aren't the equivalent internal helpers deleted or at least pruned?

Reply:
the internal helpers stays where they were, they're not pruned under k8s.io/apiextensions-apiserver/pkg/apis/apiextensions/helpers.go 

my IDE tooling tells me that we can cleanup some of those, do you think we should clean them up as well?

Reply:
yes please